arangod/RestHandler/RestDocumentHandler.cpp

```diff
@@ -203,6 +203,9 @@ RestStatus RestDocumentHandler::insertDocument() {
   if (!isMultiple && !opOptions.isOverwriteModeUpdateReplace()) {
     _activeTrx->addHint(transaction::Hints::Hint::SINGLE_OPERATION);
   }
+  if (!opOptions.isSynchronousReplicationFrom.empty()) {
```

Comment:
this might be not empty on a single-server as well no?

Reply:
I think it is only non-empty if this is a follower in a cluster. If it is non-empty on a single server then there are no further consequences, even if the hint is set for the transaction.